src/mrdsrg-spin-integrated/mrdsrg_nonpt.cc

```diff
@@ -512,23 +527,26 @@ void MRDSRG::compute_hbar_sequential_rotation() {
     TIME_LINE(O1_["pq"] = Hbar1_["pq"]);
     TIME_LINE(O1_["PQ"] = Hbar1_["PQ"]);
     if (eri_df_) {
-        TIME_LINE(O2_["pqrs"] = B["gpr"] * B["gqs"]);
-        TIME_LINE(O2_["pqrs"] -= B["gps"] * B["gqr"]);
-
-        TIME_LINE(O2_["pQrS"] = B["gpr"] * B["gQS"]);
-
-        TIME_LINE(O2_["PQRS"] = B["gPR"] * B["gQS"]);
-        TIME_LINE(O2_["PQRS"] -= B["gPS"] * B["gQR"]);
-//        TIME_LINE(O2_["pqrs"] = B["prg"] * B["qsg"]);
-//        TIME_LINE(O2_["pqrs"] -= B["psg"] * B["qrg"]);
-
-//        TIME_LINE(O2_["pQrS"] = B["prg"] * B["QSg"]);
-
-//        TIME_LINE(O2_["PQRS"] = B["PRg"] * B["QSg"]);
-//        TIME_LINE(O2_["PQRS"] -= B["PSg"] * B["QRg"]);
-        TIME_LINE(Hbar2_["pqrs"] = O2_["pqrs"]);
-        TIME_LINE(Hbar2_["pQrS"] = O2_["pQrS"]);
-        TIME_LINE(Hbar2_["PQRS"] = O2_["PQRS"]);
+        TIME_LINE(Hbar2_["pqrs"] = B["gpr"] * B["gqs"]);
+        TIME_LINE(Hbar2_["pqrs"] -= B["gps"] * B["gqr"]);
+
+        TIME_LINE(Hbar2_["pQrS"] = B["gpr"] * B["gQS"]);
+
+        TIME_LINE(Hbar2_["PQRS"] = B["gPR"] * B["gQS"]);
+        TIME_LINE(Hbar2_["PQRS"] -= B["gPS"] * B["gQR"]);
+
```

Comment:
Hey Sam, great work! One thing for optimization: if we want to store O2 with at most two virtual indices, it might be better to store all those blocks from the very beginning and only density fitted the bare H with three and four virtual indices. As such we do not contract B for each term, although you might want to time which one is faster.

Reply:
Thank you for the suggestion! That's a good idea. I am actually working on V_{"gggg"}, which actually can be one way to do it.